tests/utils.go

```diff
@@ -540,12 +540,13 @@ func NewRandomVirtualMachineInstanceWithBlockDisk(imageUrl, namespace string, ac
 	return NewRandomVirtualMachineInstanceWithDisk(imageUrl, namespace, sc, accessMode, k8sv1.PersistentVolumeBlock)
 }
 
-func NewRandomVMI() *v1.VirtualMachineInstance {
+func NewRandomVMI(opts ...libvmi.Option) *v1.VirtualMachineInstance {
```

Comment:
I think I understand why you do this, but there's a downside: you may motivate people to keep using this function instead of just replacing it with the relevant `libvmi` calls.

Reply:
I agree, but I would be fine with just a promise to continue this in a follow up to fix it.

My stronger concern here is that it is misleading.
The options are added to the VMI, but later several options are overridden. This is wrong.

Maybe you can try and remove in a dedicated commit:
```go
        vmi.ObjectMeta.Namespace = testsuite.GetTestNamespace(vmi)
	vmi.Spec.Domain.Resources.Requests = k8sv1.ResourceList{}
	if checks.IsARM64(testsuite.Arch) {
		// Cirros image need 256M to boot on ARM64,
		// this issue is traced in https://github.com/kubevirt/kubevirt/issues/6363
		vmi.Spec.Domain.Resources.Requests[k8sv1.ResourceMemory] = resource.MustParse("256Mi")
	} else {
		vmi.Spec.Domain.Resources.Requests[k8sv1.ResourceMemory] = resource.MustParse("128Mi")
	}
```

Reply:
I opted to drop this change and switch using `libvmi` calls as @dankenigsberg suggested